libs/rtefsutils/src/RteFsUtils.cpp

```diff
@@ -269,15 +270,41 @@ bool RteFsUtils::SetFileReadOnly(const string& path, bool bReadOnly) {
  */
 
   error_code ec;
-  fs::perms perms;
+  constexpr fs::perms write_mask = fs::perms::owner_write | fs::perms::group_write | fs::perms::others_write;
 
-  // Set file permissions
   if (bReadOnly) {
-    perms = fs::perms::owner_read | fs::perms::group_read | fs::perms::others_read;
+    // Remove write bits
+    fs::permissions(fs::path(path), write_mask, fs::perm_options::remove | fs::perm_options::nofollow, ec);
   } else {
-    perms = fs::perms::all;
+    // Inherit write bits from parent directory
+    fs::perms perms = fs::status(fs::path(path).parent_path(), ec).permissions() & write_mask;
```

Comment:
Should the function return `false` if `fs::status` returns error?

Reply:
Probably yes.